ios/HeapSettings.bundle/generate_settings_common.rb

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
+
+require 'fileutils'
+
+def perform_substitution(dir, app_id_dev, app_id_prod)
+  template = File.read File.join(dir, 'HeapSettings.plist.template')
+  output_path = File.join dir, 'HeapSettings.plist'
+
+  settings = template
+    .gsub('__HEAP_APP_ID_DEV__', app_id_dev || '')
+    .gsub('__HEAP_APP_ID_PROD__', app_id_prod || '')
+
+  IO.write output_path, settings
+end
+
+def get_heap_settings_from_json(json)
+  dev, prod, default = json.values_at('dev', 'prod', 'default')
```

Comment:
OK, been spinning on this a bunch because I'd like to figure out a way to do this that's not too messy, but the way it's structured makes it complicated.  It's not good enough to say something like `app_id_dev = (dev and dev['heapAutoInit'] and dev['heapAppId']) or (default ...` because that `heapAutoInit` key might legitimately be false... which is weird... it seems like we shouldn't have even given you the option if the file was there at all, but what can you do? (could probably easily deprecate or change without pissing off too many people I bet...)

I'm thinking the easiest thing to do here is just to check if they're `nil` right off the top and then substitute empty dictionaries if they are.  Something like:
```suggestion
dev = json.values_at('dev') or {}
prod = json.values_at('prod') or {}
default = json.values_at('default') or {}
```

... then lines 27 and 28 go back to their original formulation.  What do you think?

Reply:
Ah, I spent the better part of an hour trying to figure out how best to simplify this logic, but I hadn't even considered this.  This seems like the way to go